editor/components/TableMap.tsx

```diff
@@ -71,22 +74,57 @@ export const TableMap = ({ mapRef, geojson, mapConfig }: TableMapProps) => {
     }
   }, [geojsonBounds, mapRef]);
 
+  // todo, check if this is the right type to use
+  const [selectedFeature, setSelectedFeature] = useState<GeoJSONFeature | null>(
+    null
+  );
+  const [cursor, setCursor] = useState("grab");
+
+  const onMouseEnter = useCallback((e: MapMouseEvent) => {
+    setCursor("pointer");
+  }, []);
+
+  const onMouseLeave = useCallback(() => {
+    setCursor("grab");
+  }, []);
+
   return (
     <MapGL
       ref={mapRef}
       initialViewState={initialViewState}
       {...DEFAULT_MAP_PARAMS}
       cooperativeGestures={true}
+      onMouseEnter={onMouseEnter}
+      onMouseLeave={onMouseLeave}
+      cursor={cursor}
       onLoad={(e) => e.target.resize()}
       maxZoom={21}
+      interactiveLayerIds={["points-layer"]} // layer id defined in mapConfig
+      onClick={(e) => {
+        e.originalEvent.stopPropagation();
+        if (e.features?.length) {
+          setSelectedFeature(e.features[0]);
+        } else {
+          setSelectedFeature(null);
+        }
+      }}
     >
       <FullscreenControl position="bottom-right" />
       <NavigationControl position="top-right" showCompass={false} />
       {/* custom geojson source and layer */}
       <Source id="custom-source" type="geojson" data={geojson}>
-        <Layer id="custom-layer" type="circle" {...mapConfig?.layerProps} />
+        <Layer type="circle" {...mapConfig?.layerProps} />
       </Source>
       <MapFitBoundsControl mapRef={mapRef} bounds={geojsonBounds} />
+      {selectedFeature && (
+        <PopupWrapper
+          longitude={selectedFeature?.properties?.longitude}
+          latitude={selectedFeature?.properties?.latitude}
+          featureProperties={selectedFeature.properties}
+          PopupContent={TableMapPopupContent}
```

Comment:
probably me being old school but i was expecting the popup wrapper to use `children` to render the content. it doesn't change the functionality at all!